src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -70,17 +70,17 @@ internal abstract class ODataWriterCore : ODataWriter, IODataOutputInStreamError
                 resourceType = this.outputContext.EdmTypeResolver.GetElementType(navigationSource);
             }
 
-            ODataUri odataUri = outputContext.MessageWriterSettings.ODataUri.Clone();
+            ODataUriSlim odataUri = new ODataUriSlim(outputContext.MessageWriterSettings.ODataUri);

```

Comment:
What happens when `outputContext.MessageWriterSettings.ODataUri` is null?

Reply:
`ODataMessageWriterSettings.ODataUri` getter initializes a new instance of `ODataUri` if none was set. So, it can't return null. The old code also did not check for null at this site.

Also, in case it were null here, it's probably better to just fail than try to proceed with a null URL since most of `ODataWriterCore` expect an `ODataUri` to be present.